docs/scripts/generate-hardware-matrix.py

```diff
@@ -48,7 +48,9 @@ def main() -> None:
         if hw_requirement == 'arch':
             continue
 
-        matrix[hw_requirement] = {method: (False, int) for method in known_methods}
+        matrix[hw_requirement] = {method: (False, int) for method in known_methods}  # noqa: C420
+        # C420: current implementation creates a new tuple for each method
+        # https://github.com/teemtee/tmt/pull/3662#discussion_r2040669353
```

Comment:
I find it weird to have comments after the commented code, rather before

```suggestion
        # C420: current implementation creates a new tuple for each method
        # https://github.com/teemtee/tmt/pull/3662#discussion_r2040669353
        matrix[hw_requirement] = {method: (False, int) for method in known_methods}  # noqa: C420
```